core/block_svg.js

```diff
@@ -632,7 +632,8 @@ Blockly.BlockSvg.prototype.createTabList_ = function() {
   var list = [];
   for (var i = 0, input; input = this.inputList[i]; i++) {
     for (var j = 0, field; field = input.fieldRow[j]; j++) {
-      if (field instanceof Blockly.FieldTextInput && field.isVisible()) {
+      if (Blockly.FieldTextInput &&
+          field instanceof Blockly.FieldTextInput && field.isVisible()) {
```

Comment:
This smells pretty bad, and the TODO on the next line confirm that this is the wrong approach since more and more subclass code needs to be injected here if we continue this approach.  Suggest moving this code to the subclasses.